src/twisted/conch/test/test_filetransfer.py

```diff
@@ -855,6 +855,9 @@ def test_constantsAgainstSpec(self):
             self.assertEqual(v, getattr(filetransfer, k))
 
 
+# We don't run on Windows, as we don't have an SFTP file server implemented in conch.ssh for Windows.
+# As soon as there is such an implementation, we can run these tests on Windows.
+@skipIf(not unix, "can't run on non-posix computers")
```

Comment:
I'm surprised about this.  The SFTP code is all quite old.  I wonder why these tests weren't already skipped or didn't fail in the past on Windows CI.

Reply:
You added this test here https://github.com/twisted/twisted/pull/1685

and the Azure tests were not execute for that CI.


The issue is with `filetransfer.FileTransferServer(avatar=TestAvatar())`
which will try to adapt TestAvatar` into an `ISFTPServer` but the `components.registerAdapter` for `TestAvatar` is only called on Linux.